packages/dataviews/src/view-list.js

```diff
@@ -89,6 +93,19 @@ export default function ViewList( {
 										</div>
 									</VStack>
 								</HStack>
+								{ onDetailsChange && (
+									<Button
```

Comment:
I'm looking at how to make this button appear _visually_ within the other interactive element (`<div role="button">`) while avoid nesting in the DOM.

Reply:
I played with the idea of having two buttons: one nested and hidden from the accessibility tree, the second not nested and visually hidden.

It works, as demonstrated in the accessibility tree. Though, I'm not super keen on this idea:

https://github.com/WordPress/gutenberg/assets/583546/0f317579-d323-4589-b84d-f48d10fea06e

<details>
<summary>In terms of code, something along these lines could work. It'd need factoring into a single component.</summary>
```diff
diff --git a/packages/dataviews/src/style.scss b/packages/dataviews/src/style.scss
index 93926db47d..72037da3d6 100644
--- a/packages/dataviews/src/style.scss
+++ b/packages/dataviews/src/style.scss
@@ -280,6 +280,16 @@
        .dataviews-list-view__details-button {
                align-self: center;
        }
+       .dataviews-list-view__details-button-visually-hidden {
+                       position: absolute;
+                       width: 1px;
+                       height: 1px;
+                       margin: -1px;
+                       padding: 0;
+                       overflow: hidden;
+                       clip: rect(0, 0, 0, 0);
+                       border: 0;
+       }
 }
 
 .dataviews-action-modal {
diff --git a/packages/dataviews/src/view-list.js b/packages/dataviews/src/view-list.js
index cf9ca31851..ccfc821956 100644
--- a/packages/dataviews/src/view-list.js
+++ b/packages/dataviews/src/view-list.js
@@ -95,6 +95,7 @@ export default function ViewList( {
                                                                </HStack>
                                                                { onDetailsChange && (
                                                                        <Button
+                                                                               aria-hidden={true}
                                                                                className="dataviews-list-view__details-button"
                                                                                onClick={ () =>
                                                                                        onDetailsChange( [ item ] )
@@ -108,6 +109,19 @@ export default function ViewList( {
                                                                ) }
                                                        </HStack>
                                                </div>
+                                               { onDetailsChange && (
+                                                                       <Button
+                                                                               className="dataviews-list-view__details-button-visually-hidden"
+                                                                               onClick={ () =>
+                                                                                       onDetailsChange( [ item ] )
+                                                                               }
+                                                                               icon={
+                                                                                       isRTL() ? chevronLeft : chevronRight
+                                                                               }
+                                                                               label={ __( 'View details' ) }
+                                                                               size="compact"
+                                                                       />
+                                                               ) }
                                        </li>
                                );
                        } ) }

````
</details>

Reply:
I've also played with them being siblings within the same HStack. Tried different approaches, but wasn't able to make the first interactive element take all the space up to the second. There's always a weird gap.

<img width="341" alt="Captura de ecrã 2024-01-05, às 12 28 38" src="https://github.com/WordPress/gutenberg/assets/583546/3579f44f-6777-4e94-a46d-36506dbed5ce">

Even if we were able to remove that gap and the button take the full height, visually, they will appear as two consecutive elements (see focus). Would that be a fine trade-off design-wise @jameskoster ?

https://github.com/WordPress/gutenberg/assets/583546/3412ec11-291c-4f39-9312-72d8e8bf7b23

Reply:
cc @andrewhayward

Reply:
Instead of duplicating 'details' buttons, could the main button and details button be adjacent, with the latter positioned absolutely work? Something like [this codepen](https://codepen.io/jameskoster/pen/KKEVOqR).

Reply:
> Something like [this codepen](https://codepen.io/jameskoster/pen/KKEVOqR).

That's brilliant, thanks! They are siblings in the accessibility tree, indeed:

<img width="281" alt="Captura de ecrã 2024-01-05, às 15 06 21" src="https://github.com/WordPress/gutenberg/assets/583546/1969e770-c5bf-4e98-adad-2149ec672e7a">

I'll implement that.

Reply:
> I've also played with them being siblings within the same HStack. Tried different approaches, but wasn't able to make the first interactive element take all the space up to the second. There's always a weird gap.

Did you set `spacing` to `0`?

> Instead of duplicating 'details' buttons, could the main button and details button be adjacent, with the latter positioned absolutely?

Modifying @jameskoster's [codepen](https://codepen.io/jameskoster/pen/KKEVOqR) slightly should work with `flex`, and avoid potentially confusing tap targets...

<img src="https://github.com/WordPress/gutenberg/assets/159848/8da92fad-c7a1-4de7-ab75-1ffeb2fa2e25" alt="" height="150px" align="left" />

```css
li {
  display: flex;
}

.row {
  flex-grow: 1;
}
```

Reply:
@andrewhayward Ideally the first button spans the entire width of the container so that the hover and focus styles remain as they are. 

Separating the buttons with no overlap while maintaining alignment gets a bit tricky and adds some visual noise:

![details](https://github.com/WordPress/gutenberg/assets/846565/855aaf4d-e32f-4fb7-bf6c-7895287c8f35)

I'm not sure the trade-off is worth it?

Reply:
Speaking of visual noise, it would be good to try revealing the details button only when either:

1. The row is hovered.
2. It is focussed.

Reply:
> Ideally the first button spans the entire width of the container so that the hover and focus styles remain as they are.

We could use `grid` rather than `flex` to do that, particularly if we want to make sure that items are otherwise aligned...

![Hover and focus effects on overlapping buttons in a list](https://github.com/WordPress/gutenberg/assets/159848/31e78365-5876-4164-9bd6-42b4cda22499)

That being said, if the background of the whole item changes when the primary action is hovered/focused, does that obscure the fact that there are two actions? It doesn't say to me immediately that the chevron is independent, and is not super discoverable.

> ...it would be good to try revealing the details button only when...

Can you explain the benefit of doing this a bit more? I'm not sure how noisy they are, but hiding them would potentially cause issues.

Reply:
Related, if the entire element is going to be a button, we'll have to be a bit more conscious of what the accessible names look like. By default, it's the text content of the control, which right now would probably be something like "Sample Page admin published", which isn't great. The best way to fix this would probably be by setting `aria-labelledby` to point at the `primaryField` content.

That being said, the main job of an accessible name is to convey the purpose or intent of the element, and just using the primary field value doesn't really do that. So we'll have to think a bit about the semantics around this; a straight-up list of buttons might not actually be the best model here anyway.

Reply:
> Can you explain the benefit of doing this a bit more?

For sighted users the visual noise of the constantly-visible chevron is a bit distracting. It feels overly prominent in the UI. Perhaps a bad example but I'd liken it to tabs in a web/file browser, IDE, etc., where the `close` button is generally not permanently visible.

If the overlapping button is going to be problematic perhaps we can try a different hover/focus treatment of the main button. I'll try a design iteration tomorrow.

Reply:
@andrewhayward it occurred to me that the pattern I'm describing here is virtually identical to List View in the Editor. While not perfect, it has been quite rigorously tested. Do you think that since there's a precedent we might mimic that pattern here?

Reply:
Here's what I have:

https://github.com/WordPress/gutenberg/assets/583546/58f93148-b44d-4b27-b04d-f6e5decc9c1c

Implemented it following the block editor list view pattern:

- structurally, buttons are siblings
- most styles are handled by the button's wrapper 
- the focus styles are attached to a `::after` pseudo-element

Reply:
Looking at how the block editor list view works from a keyboard perspective, a follow-up to this PR could be to make dataviews' list view work the same: tab only stops at the top-most item of the list layout, people use arrow keys to navigate through the items and accessing the details.